Source/Core/Core/PrimeHack/Mods/FpsControls.cpp

```diff
@@ -516,50 +486,63 @@ void FpsControls::CheckBeamVisorSetting(Game game, Region region)
   switch (game) {
   case Game::PRIME_1: {
     if (region == Region::NTSC_U) {
-      beam_value = beam ?  0x4BFFFCBD: 0x60000000;
-      visor_value = visor ? 0x4BFFFCD5 : 0x60000000;
-      beam_address = 0x80075F24;
-      visor_address = 0x80075F0C;
+      beam_value = beam ?  0x4bfffcBd: 0x60000000;
+      visor_value = visor ? 0x4bfffcd5 : 0x60000000;
+      beam_address = 0x80075f24;
+      visor_address = 0x80075f0c;
     }
     else if (region == Region::PAL) {
-      beam_value = beam ?  0x4BFFFCD5: 0x60000000;
-      visor_value = visor ? 0x4BFFFCBD : 0x60000000;
-      beam_address = 0x80075F74;
-      visor_address = 0x80075F8C;
+      beam_value = beam ?  0x4bfffcd5: 0x60000000;
+      visor_value = visor ? 0x4bfffcBd : 0x60000000;
+      beam_address = 0x80075f74;
+      visor_address = 0x80075f8c;
+    }
+    else { // NTSC-J
+      beam_value = beam ? 0x4bfffcBd : 0x60000000;
+      visor_value = visor ? 0x4bfffcd5 : 0x60000000;
+      beam_address = 0x800760a4;
+      visor_address = 0x8007608c;
     }
-    else {} // NTSC-J
     break;
   }
   case Game::PRIME_2: {
     if (region == Region::NTSC_U) {
-      beam_value = beam ?  0x4BFFFC7D: 0x60000000;
-      visor_value = visor ? 0x4BFFFC99 : 0x60000000;
-      beam_address = 0x8006FDE0;
-      visor_address = 0x8006FDC4;
+      beam_value = beam ?  0x4bfffc7d: 0x60000000;
+      visor_value = visor ? 0x4bfffc99 : 0x60000000;
+      beam_address = 0x8006fde0;
+      visor_address = 0x8006fdc4;
     }
     else if (region == Region::PAL) {
-      beam_value = beam ?  0x4BFFFC7D: 0x60000000;
-      visor_value = visor ? 0x4BFFFC99 : 0x60000000;
+      beam_value = beam ?  0x4bfffc7d: 0x60000000;
+      visor_value = visor ? 0x4bfffc99 : 0x60000000;
       beam_address = 0x80071358;
-      visor_address = 0x8007133C;
+      visor_address = 0x8007133c;
+    }
+    else { // NTSC-J
+      beam_value = beam ? 0x4bfffc7d : 0x60000000;
+      visor_value = visor ? 0x4bfffc99 : 0x60000000;
+      beam_address = 0x8006fbb0;
+      visor_address = 0x8006fb94;
     }
-    else {} // NTSC-J
     break;
   }
   case Game::PRIME_3: {
     // NTSC and PAL are the same here. Seriously, I double checked.
     visor_value = visor ? 0x41820018 : 0x48000018;
-    visor_address = 0x800614EC;
+    visor_address = 0x800614ec;
     break;
   }
   case Game::PRIME_3_STANDALONE: {
+    visor_value = visor ? 0x41820018 : 0x48000018;
     if (region == Region::NTSC_U) {
-
+      visor_address = 0x800617e4;
     }
     else if (region == Region::PAL) {
-
+      visor_address = 0x80061958;
+    }
+    else { // NSTC-J
+      visor_address = 0x80061a88;
     }
```

Comment:
Are you certain this is correct? The value written is either NOP or a branch to the beam/visor function. The distance between the trigger to open the visor menu and the visor menu func address is identical for prime 3 trilogy, prime standalone ntsc, pal and JP according to this. Any difference in size of the functions between versions would break the call.